packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -628,25 +629,32 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
 
   // SCROLL WHEEL
 
+  // Returns the offset that should result from applying [event] to the current
+  // position, taking min/max scroll extent into account.
+  double _targetScrollOffsetForPointerScroll(PointerScrollEvent event) {
+    final double delta = _pointerSignalEventDelta(event);
+    return math.min(math.max(position.pixels + delta, position.minScrollExtent),
+      position.maxScrollExtent);
+  }
+
   // Returns the delta that should result from applying [event] with axis and
   // direction taken into account.
-  double _targetScrollDeltaForPointerScroll(PointerScrollEvent event) {
+  double _pointerSignalEventDelta(PointerScrollEvent event) {
     double delta = widget.axis == Axis.horizontal
-        ? event.scrollDelta.dx
-        : event.scrollDelta.dy;
+      ? event.scrollDelta.dx
+      : event.scrollDelta.dy;
 
     if (axisDirectionIsReversed(widget.axisDirection)) {
       delta *= -1;
     }
-
     return delta;
   }
 
   void _receivedPointerSignal(PointerSignalEvent event) {
     if (event is PointerScrollEvent && _position != null) {
-      final double targetScrollOffset = _targetScrollDeltaForPointerScroll(event);
+      final double targetScrollOffset = _targetScrollOffsetForPointerScroll(event);
       // Only express interest in the event if it would actually result in a scroll.
-      if (targetScrollOffset != 0) {
+      if (targetScrollOffset != position.pixels) {
```

Comment:
Should we be checking `_physics != null && !_physics!.shouldAcceptUserOffset(position))` before calling `pointerSignalResolver.register`?

Reply:
I think that's right now because #37211

Reply:
That was part of another change, but I moved it to precede `pointerSignalResolver.register` and all the tests passed. Makes sense to me!

Reply:
```dart
              Stack(
                children: [
                  Positioned.fill(
                    child: someList(),
                  ),
                  someList(
                      physics:NeverScrollableScrollPhysics,
                  ),
                ],
              );
```

If my pointer signal scrolls on the second list, should the expected behavior be to scroll nothing at all?

If you moved, then the first list is scrolled.

I believe this is a bit counterintuitive?

Reply:
This is actually a common design pattern I've seen before. In the repro provided in the bug, it creates two parallel scroll views that scroll as one. Another similar use case was reported in https://github.com/flutter/flutter/issues/70902